src/main/webapp/resources/js/pages/projects/samples-metadata-import/redux/importReducer.ts

```diff
@@ -91,79 +95,173 @@ export const saveMetadata = createAsyncThunk<
-      const promises: Promise<void>[] = [];
-      for (let j = i; j < i + chunkSize && j < metadata.length; j++) {
-        const metadataItem: MetadataItem = metadata[j];
-        const index: string = metadataItem.rowKey;
-        if (
-          selectedMetadataKeys.includes(index) &&
-          metadataSaveDetails[index]?.saved !== true
-        ) {
-          const name: string = metadataItem[sampleNameColumn];
-          const metadataFields: FieldUpdate[] = Object.entries(metadataItem)
-            .filter(
-              ([key]) =>
-                headers.map((header) => header.name).includes(key) &&
-                key !== sampleNameColumn
-            )
-            .map(([key, value]) => ({
-              field: key,
-              value,
-              restriction: headers.filter((header) => header.name === key)[0]
-                .targetRestriction,
-            }));
-          const sampleId = metadataValidateDetails[index].foundSampleId;
-          if (sampleId) {
-            promises.push(
-              updateSample({
-                projectId,
-                sampleId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          } else {
-            promises.push(
-              createSample({
-                projectId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          }
-        }
-      }
-      await Promise.all(promises).then(() => {
-        dispatch(
-          setMetadataSaveDetails(Object.assign({}, metadataSaveDetails))
+    const selectedSampleList = metadata.filter((metadataItem) => {
+      const name: string = metadataItem[sampleNameColumn];
+      return (
+        selectedMetadataKeys.includes(metadataItem.rowKey) &&
+        metadataSaveDetails[name]?.saved !== true
+      );
+    });
+
+    //update existing project samples
+    const updateSamplesPromises: Promise<void>[] = [];
+    const updateSampleList = selectedSampleList
+      .filter((metadataItem) => {
+        const name = metadataItem[sampleNameColumn];
+        const sampleId = metadataValidateDetails[name].foundSampleId;
+        return sampleId;
```

Comment:
Can you make this more explicit you could do `return typeof sampleId === 'number'`

Reply:
Please see 54cc39e2e25d01d27e572c0b363f4b3cd8b64344.